### PR TITLE
Fix OIDC issuer ID in resident IdP

### PR DESCRIPTION
**Pull request by ChamathNS**
### Proposed changes in this pull request

In tenant-qualified URLs mode, the OIDC Identity Provider Entity ID would be set to the token EP and would not be configurable.

-

### When should this PR be merged

ASAP

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 1]
```diff
@@ -206,6 +206,7 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         if (StringUtils.isBlank(oauth2TokenEPUrl)) {
             oauth2TokenEPUrl = IdentityUtil.getServerURL(IdentityConstants.OAuth.TOKEN, true, false);
```
Comment (ashensw) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:207]: Are we still using the previous util method as well? Can't we use the new ServiceURLBuilder hear as well?

Comment (ChamathNS) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:207]: Yes. We can change these to use the ServiceURLBuilder.

Comment (ChamathNS) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java]: The PR is now updated.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 2]
```diff
@@ -2515,17 +2513,31 @@ private void validateResourceId(String resourceId, String tenantDomain) throws I
...
-    private String resolveAbsoluteURL(String urlContext, String resolvedUrl) throws IdentityProviderManagementServerException {
+    /**
+     * Resolves the public service url given the default context and the url picked from the configuration based on
+     * the 'tenant_context.enable_tenant_qualified_urls' mode set in deployment.toml.
+     *
+     * @param defaultUrlContext default url context path
+     * @param urlFromConfig     url picked from the file configuration
+     * @return absolute public url of the service if 'enable_tenant_qualified_urls' is 'true', else returns the url
+     * from the file config
+     * @throws IdentityProviderManagementServerException when fail to build the absolute public url
```
Comment (janakamarasena) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:2524]: Start sentence with a capital letter.

Comment (ChamathNS) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java:2524]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3064.

**Pull request merged**